lagotto/init.sls

```diff
@@ -1,123 +1,69 @@
-
-extend:
-  apt-repo-plos:
-    pkgrepo.managed:
-      - require_in:
-        - pkg: lagotto-apt-packages
-
-lagotto-service:
-  service.running:
-    - name: lagotto
-    - enable: true
-    - require:
-      {%- if oscodename == 'trusty' %}
-      - file: /etc/init/lagotto.conf
-      {%- else %}
-      - file: /etc/systemd/system/lagotto.service
-      {%- endif %}
-    - watch:
-      - file: {{ app_root }}/shared/puma.rb
-
-lagotto-apt-packages:
-  pkg.installed:
-    - pkgs:
-        - build-essential
-        - libgmp-dev
-        - libmysqlclient-dev
-        - libssl-dev
-        - nodejs
-        {% if oscodename == 'bionic' %}
-        - node-gyp
-        - nodejs-dev
-        - libssl1.0-dev
-        - npm
-        {% endif %}
-
-{{ app_root }}:
-  file.directory:
-    - user: lagotto
-    - group: lagotto
-    - require:
-      - user: lagotto
-
-{{ app_root }}/shared:
-  file.directory:
-    - user: lagotto
-    - group: lagotto
-    - require:
-      - file: {{ app_root }}
-
-{{ app_root }}/shared/puma.rb:
-  file.managed:
-    - template: jinja
-    - source: salt://lagotto/etc/puma/puma.rb
-    - require:
-      - file: {{ app_root }}/shared
-
-{%- if oscodename == 'trusty' %}
-/etc/init/lagotto.conf:
-  file.managed:
-    - template: jinja
-    - source: salt://lagotto/etc/init/lagotto.conf
-    - user: root
-    - group: root
-    - mode: 644
-{%- else %}
-/etc/systemd/system/lagotto.service:
-  file.managed:
-    - template: jinja
-    - source: salt://lagotto/etc/systemd/system/lagotto.service
-{%- endif %}
-
-/etc/sudoers.d/lagotto:
-  file.managed:
-    - template: jinja
-    - source: salt://lagotto/etc/sudoers.d/lagotto
+  - lagotto.lib.docker-common
 
-/etc/logrotate.d/rails:
-  file.managed:
-    - template: jinja
-    - source: salt://lagotto/etc/logrotate.d/rails
+{% set environment = salt.grains.get('environment') %}
+{% set app_port = props.get('app_port') %}
+{% set mysql_host = consul_service_domain("alm-manager-sql", style='soma') %}
 
-lagotto-sysctl-ip-local-port-range:
-  sysctl.present:
-    - name: net.ipv4.ip_local_port_range
-    - value: {{ ip_local_port_range }}
-
-{%- if oscodename == 'trusty' %}
-lagotto-sysctl-tcp-tw-recycle:
-  sysctl.present:
-    - name: net.ipv4.tcp_tw_recycle
-    - value: {{ tcp_tw_recycle }}
-{%- endif %}
-
-lagotto-sysctl-tcp-tw-reuse:
-  sysctl.present:
-    - name: net.ipv4.tcp_tw_reuse
-    - value: {{ tcp_tw_reuse }}
-
-/usr/local/bin/rake:
-  file.symlink:
-    - target: /opt/rubies/ruby-{{ ruby_ver }}/bin/rake
+{{ app_name }}-app-container-absent:
+  docker_container.absent:
+    - onchanges:
+      - {{ app_name }}-image
     - force: True
+    - names:
+      - {{app_name}}-app
+
+{{ app_name }}-network:
+  docker_network.present:
+    - name: {{ app_name }}
+
+{{ app_name }}-app-container-running:
+  docker_container.running:
+    - name: {{ app_name }}-app
+    - image: {{ docker_image_name }}
+    - environment:
+{%- for name, value in props.items() %}
```

Comment:
Simple translation!  I like the idea of expecting salt props to be exposed as env vars.

Reply:
Yeah! I wasn't quite sure how controversial this idea would be. It sure makes these states cleaner and easier to work with, but it comes with the downside that _everything_ in this dict will be made available to the docker container, even stuff it doesn't strictly need to know about.

Reply:
A whitelist would be simple too, though it doesn't seem controversial enough with lagotto to warrant it.